NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -150,121 +242,157 @@ NFIQ2UI::executeSingle(std::shared_ptr<BE::Image::Image> img,
 
 	cv::Mat postResample {};
 
-	if ((resolution.xRes != resolution.yRes) || (imageDPI != 500)) {
-		// Possible re-sampling
-		if (flags.force) {
+	static const uint8_t defaultDPI { 72 };
+	static const uint16_t requiredDPI { 500 };
+
+	const NFIQ2UI::ResampleDims resampleDims { imageHeight, imageWidth,
+		imageDPI, requiredDPI };
+	const NFIQ2UI::ImageProps imageProps { name, fingerPosition, quantized,
+		resampled, singleImage };
```

Comment:
If you're going to use this struct, use it all the time. You now have `resampled` (non-`const`) and `imageProps.resampled` (`const`), etc. You _will_ screw this up. I'm not even going to check the usage here.

Make this non-const and declare it at the top.